### PR TITLE
enforce spaces when indentation detected

### PR DESCRIPTION
**Pull request by kevinushey**
Fixes part of https://github.com/rstudio/rstudio/issues/4448.

### CODE REVIEW

**Comment by jmcphers** [comment]
Do we need this for 1.2?

**Comment by kevinushey** [comment]
What's your take? For reference, this is the specific issue this fix is referencing:

---

Perhaps this is what you're seeing... Reproduction steps:

1. Create a new RStudio project.
2. Open the project options. Uncheck 'Insert spaces for tab', and set the tab width to 8.
3. Create a new file (<kbd>Ctrl + Shift + N</kbd>) and insert a tab. It should be 8 spaces, as expected.
4. Try running this R code:

```r
text <- rep("  indent", 10)
writeLines(text, con = "file.R")
file.edit("file.R")
```

Go to the bottom of the file, and try inserting a tab. It will be of size 2 rather than 8.

In other words, the issue appears to affect only _existing_ documents, not new ones.

---

Technically this is a regression, and can be worked around relatively easily. But the fix itself is minor and should be safe.

**Comment by jmcphers** [comment]
Okay, I'm sold since it's a regression. Thanks for fixing this!
